ingest-app/src/cmr/ingest/validation/validation.clj

```diff
@@ -94,16 +94,41 @@
      :ISOTopicCategories (match-kms-keywords-validation
                           kms-index :iso-topic-categories msg/iso-topic-category-not-matches-kms-keywords)}))
 
+(defn- log-full-error-message
+  "Log entire validation error message so that information is not lost."
+  [error]
+  (info (str "UMM Validation error. Full message: " error)))
+
+(defn- create-user-friendly-error-message
+  "Remove non user-friendly things like regexes from error message to be returned.
+   The original message should be logged."
+  [error]
+  (let [friendly-error (string/split error #";")]
+    (if (>= (count friendly-error) 1)
+      (do
+       (log-full-error-message error)
+       (first friendly-error))
+      error)))
+
+(defn- sanitize-error-messages
```

Comment:
I couldn't tell what this function does to the error message. It would be nice to have an example of the before and after message in the docstring or test to make sure that we indeed made the error message more informative.